### PR TITLE
@jmchilton's GCC Training Homework

### LINKED ISSUE #231: Dev Corner - Conda for Tool Dependencies
There is a good chance "Conda and Containers for Tool Dependencies" will be selected for a tutorial at the GCC. It would be good to have a dedicated tutorial to building up Conda dependencies with a Galaxy and planemo-flavored approach.

Should pair with additional documentation for planemo and link to relevant bioconda tutorials. Will likely require updates to planemo.

### LINKED ISSUE #232: Dev Corner - Docker for Tool Dependencies
There is a good chance "Conda and Containers for Tool Dependencies" will be selected for a tutorial at the GCC. It would be good to have a dedicated tutorial for using Docker as tool dependencies. It would be nice if it covered two approaches - manual and with BioContainers magic.

- A good existing - if somewhat dated - resource for manual Docker dependencies would be the tutorial from @apetkau at https://github.com/apetkau/galaxy-hackathon-2014.
- I don't think any BioContainers documentation exists but it should be built up (probably as part of planemo) and a set of slides added to this repository.

### PR DESCRIPTION
**Pull request by jmchilton**
Included in this PR:

- Tool development using Conda and Bioconda slides. Fixes #231.
- Tool development with Containers and BioContainers slides. Fixes #232.
- Galaxy Code Architecture slides synchronzied from galaxyproject/galaxy.

I still have a lot I'd like to do but this seems like a workable first draft.

Conda:
 - [ ] Discussion of merged mulled environments perhaps, more slides about dependency resolution in general?

Containers:
 - [x] Update all slides for mulled-v2. 
 - [x] Update with information on multi-package-containers repository and monitoring tool repositories for missing containers.
 - [ ] Singularity information.
 - [ ] Information on newer container-based deployment options - Kubernetes, Chronos, and Condor Docker universe.

Arch:
 - [ ] Redo startup slides since they may have changed since last year - especially tool monitoring and stuff.
 - [ ] Add a section about web hooks.

### CODE REVIEW

File: topics/dev/tutorials/conda/slides.html [hunk 1]
```diff
@@ -0,0 +1,644 @@
...
+![](../../../../../shared/images/conda_logo.png)
+
+Package, dependency and environment management
+
+---
+
+###.image-25[![](../../../../../shared/images/conda_logo.png)] <br/> Conda Terminology
+
+
+Conda **recipes** build **packages** that published to **channels**.
```
Comment (nsoranzo) [inline topics/dev/tutorials/conda/slides.html]: that *are* published?

File: topics/dev/tutorials/conda/slides.html [hunk 2]
```diff
@@ -0,0 +1,644 @@
...
+---
+
+###.image-25[![](../../../../../shared/images/conda_logo.png)] <br/> Channels
+
+- Packages through channels within Continuum.
+- Conda channels searched by Galaxy for packages:
+    - iuc
+    - bioconda
+    - defaults
+    - r
```
Comment (nsoranzo) [inline topics/dev/tutorials/conda/slides.html]: `r` channel should be above `defaults`

File: topics/dev/tutorials/conda/slides.html [hunk 3]
```diff
@@ -0,0 +1,644 @@
...
+### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on
+
+![](../../images/exercise.png)
+
+---
+
+### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on
+
+#### The Goal
+- Use the Planemo commands `conda_install`, `conda_env`, and `test` to learn the Galaxy tool depenendency development lifecycle.
```
Comment (nsoranzo) [inline topics/dev/tutorials/conda/slides.html]: s/depenendency/dependency/

File: topics/dev/tutorials/conda/slides.html [hunk 4]
```diff
@@ -0,0 +1,644 @@
...
+2. `cd seqtk_example`
+3. `planemo conda_install seqtk_seq.xml`
+4. `planemo conda_env seqtk_seq.xml`
+5. `planemo test seqtk_seq.xml`
+
+---
+
+### Finding the Correct Requirements & Packages
+
+The above example worked because a published Bioconda recipe named`seqtk` at version `1.2`
```
Comment (nsoranzo) [inline topics/dev/tutorials/conda/slides.html]: Missing space between "named" and "`seqtk`"

File: topics/dev/tutorials/conda/slides.html [hunk 5]
```diff
@@ -0,0 +1,644 @@
...
+Notice Planemo indicates this tool is available and shows the channel it is available in.
+
+---
+
+### The Planemo `conda_install` command
+
+```
+$ planemo conda_install seqtk_seq.xml
+Install conda target CondaTarget[seqtk,version=1.2]
+/home/john/miniconda2/bin/conda create -y --name __seqtk@1.2 seqtk=1.2
```
Comment (nsoranzo) [inline topics/dev/tutorials/conda/slides.html]: s/miniconda2/miniconda3/ here and in a few other places below.

**Comment by nsoranzo** [comment]
I think that since `topics/dev/tutorials/conda/slides.html` reuses lots of slides from `topics/dev/tutorials/tool-integration/slides.html`, it would be good to remove the Conda part from the latter, because:
- it was too big and, as discussed at the Cambridge hackathon, it would be good to split some tutorial/slide decks in smaller pieces
- having both would inevitably cause a duplication of maintenance effort.

Thoughts? Ping @lecorguille @abretaud

**Comment by jmchilton** [comment]
> it would be good to remove the Conda part from the latter

Yeah - a few of us discussed this on Gitter and that was agreed to. On my next pass I'll include those changes in this PR.

```
John Chilton @jmchilton May 30 13:01
Not for planemo - I'm making a new module inside of training-materials - I just want to reuse some of the slides from a different module (maybe module isn't the right word)
It is fine I think - I'll copy and paste for now - an ability to arbitrarily import shared slides would be cool though

Björn Grüning @bgruening May 30 13:03
mh, I think then the correct way is to split the huge slide-deck into smaller onces
we agreed to make it more modular

John Chilton @jmchilton May 30 13:11
@bgruening Fair enough point on breaking the big deck into smaller ones - I'll look at this work I'm doing now as a first step to break the Conda and Dependency Management stuff out of tool integration deck then.

Björn Grüning @bgruening May 30 13:11
Great, thanks!

Gildas Le Corguillé @lecorguille May 31 10:01
:thumbsup:
```

**Comment by lecorguille** [comment]
Has I said, I agree to extract the conda slides from the dev part.

**Comment by lecorguille** [comment]
I catched some issue with some pictures in the dev part.
I will fix them as soon as you remove the conda stuff.

**Comment by abretaud** [comment]
No problem for me :+1: :+1:

**Comment by nsoranzo** [comment]
> For me, you can cut from the slide 73 and by this way keep some references to Conda. And eventually, add a link to the new dedicated material.

Yep, till slide 95.

**Comment by nsoranzo** [comment]
@jmchilton I'm working on removing the Conda slides and other fixes to `tool-integration/slides.html`, I can push them to your branch if that's OK with you.

**Comment by jmchilton** [comment]
@nsoranzo Yes - definitely feel free to push directly to my branch thanks - that will be a huge help!

I've spent the day redoing the Planemo docs for changes in the last couple weeks and so I haven't gotten back to reviewing these slides. I will update the slides for all your inline comments as I make my way back through the slides to synchronize them with the updated Planemo docs.

Thanks for the comments all!

**Comment by lecorguille** [comment]
@jmchilton 
I plan to remove the Hands-on slides from the tool_integration slides to link the Planemo tutorial.
I will check your last version but maybe I will add some anchors along the tuto if needed.

**Comment by nsoranzo** [comment]
@lecorguille Please wait for my changes to appear in this PR tomorrow morning, I'm moving some Planemo slides, thanks!

**Comment by lecorguille** [comment]
Take your time!
I'm on holiday the evening until the GCC :)

**Comment by nsoranzo** [comment]
@jmchilton @lecorguille I pushed my changes.

@martenson You may want to take a look at this PR which has the Conda slides removed from the "Tool development and integration" tutorial.

@bebatut @bgruening It would be good to also have a review from one of you core GTN maintainers.

File: topics/dev/tutorials/conda/slides.html (hunk 4 above)
Comment (bgruening) [inline topics/dev/tutorials/conda/slides.html]: @nsoranzo do you mind to change this as well?

Comment (nsoranzo) [inline topics/dev/tutorials/conda/slides.html]: @bgruening John said "I will update the slides for all your inline comments as I make my way back through the slides to synchronize them with the updated Planemo docs.", so I'll leave them to him.

**Comment by jmchilton** [comment]
@nsoranzo - I just noticed these slides are broken like described in #358 - I don't know how to fix them. Do you know? I tried the change from that PR but applied to these files and it didn't seem to work.

**Comment by nsoranzo** [comment]
@jmchilton Broken how? I think `tutorial_slides` layout should be fine, since they have `Hands-on` sections.

**Comment by jmchilton** [comment]
The questions, objectives, and key points are not being automatically populated from the metadata anymore (they were fine before I rebased against the merged build-pipeline master branch).

File: topics/dev/tutorials/conda/slides.html (hunk 4 above)
Comment (nsoranzo) [inline topics/dev/tutorials/conda/slides.html]: @bgruening All my review comments have been kindly fixed by @jmchilton now.

File: topics/dev/tutorials/conda/slides.html [hunk 6]
```diff
@@ -591,7 +591,7 @@
 ###.image-25[![](../../../../../shared/images/bioconda_logo.png)] <br /> Contributing
 
 1. Fork [Bioconda](https://github.com/bioconda/bioconda-recipes/fork).
-1. Clone your the fork:
+1. Clone your fork:
```
Comment (jmchilton) [inline topics/dev/tutorials/conda/slides.html]: Sorry about missing that, thanks for the follow up.

Comment (nsoranzo) [inline topics/dev/tutorials/conda/slides.html]: NP

File: topics/dev/metadata.yaml [hunk 7]
```diff
@@ -232,6 +232,97 @@ material:
...
     - "A Galaxy Webhook provides an easy way to customize the Galaxy UI with plugins"
     - "Right now there are 4 entry points: tool, workflow, masthead, history-menu"
+-
+    title: "Tool Dependencies and Conda"
+    type: "tutorial"
+    name: "conda"
+    zenodo_link: ""
+    galaxy_tour: ""
+    hands_on: "external"
+    hands_on_url: "http://planemo.readthedocs.io/en/latest/writing_standalone.html"
```
Comment (bgruening) [inline topics/dev/metadata.yaml]: I will change this @jmchilton if you let me know if this is correct.

**Comment by bgruening** [comment]
I'm going to merge this so we can all work on master for smaller changes. It should now render fine!

Thanks a lot for your work, this looks awesome! To bad I can not see this in action :)

**Pull request merged**